src/callBuilder.js

```diff
@@ -0,0 +1,139 @@
+import {NotFoundError, NetworkError, BadRequestError} from "./errors";
+
+let URI = require("URIjs");
+let URITemplate = require('URIjs/src/URITemplate');
+
+
+let axios = require("axios");
+var EventSource = (typeof window === 'undefined') ? require('eventsource') : window.EventSource;
+let toBluebird = require("bluebird").resolve;
+
+/**
+* @class Builder
+*/
+export class CallBuilder {
+
+    /*
+    * @constructor
+    */
+    constructor(url) {
+        this.url = url;
+        this.filter = [];
+    }
+
+    checkFilter() {
+        if (this.filter.length >= 2) {
+            throw new BadRequestError("Too many filters specified", this.filter);
+        } 
+        if (this.filter.length === 1) {
+            this.url.segment(this.filter[0]);
+        }        
+    }
+
+    call() {
+        this.checkFilter();
+        var promise = this._sendNormalRequest(this.url)
+            .then(this._parseResponse.bind(this));
+        return promise;
+    }
+
+    /**
+    * Convert each link into a function on the response object.
+    */
+    _parseRecord(json) {
+        if (!json._links) {
+            return json;
+        }
+        var self = this;
+        var linkFn = function (link) {
```

Comment:
I'm not a fan of using the `var self = this` pattern since we have arrow functions.  Let me propose an alternative formulation:

extract `linkFn` to a private instance method name `_requestFnForLink(link)`.  This method would look like:

```js
_requestFnForLink(link) {
  return opts => {
    ...just use this like normal in here...
  }
}
```